docs/source/guides/quickstart.md

```diff
@@ -1,48 +1,195 @@
+will also be logged to the geth-running console on startup. 
+
+Connect to the Geth Console to test your ethereum node's IPC:
 ```bash
-$ nucypher ursula init --federated-only
+$ geth attach ~/.ethereum/goerli/geth.ipc
 ```
 
-### Run a Persistent Ursula
+### Wallets
+
+To list available accounts on your geth node:
 
 ```bash
-$ nucypher ursula run --network <NETWORK_DOMAIN> --teacher <SEEDNODE_URI> --federated-only
+$ geth attach ~/.ethereum/goerli/geth.ipc
+> eth.accounts
+["0x287a817426dd1ae78ea23e9918e2273b6733a43d", "0xc080708026a3a280894365efd51bb64521c45147"]
 ```
 
-Replace `<NETWORK_DOMAIN>` with the network domain and `<SEEDNODE_URI>` with the URI of a node running on that network
-domain you want to connect to (for example `0.0.0.0:9151` or `0xdeadbeef@0.0.0.0:9151`).
+Hardware wallet addresses will also be listed here if one is attached to the system hardware.
+Note that the Geth console does not return EIP-55 compliant checksum addresses, and instead will output
+the *lowercase* version of the address.  Since Nucypher requires EIP-55 checksum addresses, you will need 
+to convert your address to checksum format:
 
-If you're connecting to the `devnet`, you should use `--network devnet --teacher 18.222.119.242:9151`.
+```bash
+> web3.toChecksumAddress(eth.accounts[0])
+"0x287A817426DD1AE78ea23e9918e2273b6733a43D"
+```
 
-### Run a Geth-Connected Development Ursula
+## Connecting to The NuCypher Network
 
-Run a local geth node in development mode:
+### Connecting Nucypher to an Ethereum Provider
 
-```bash
-$ geth --dev
+```python
+from nucypher.blockchain.eth.interfaces import BlockchainInterfaceFactory
+BlockchainInterfaceFactory.initialize_interface(provider_uri='~/.ethereum/goerli/geth.ipc')
 ```
 
-Run a local development Ursula connected to the geth node
+### Ursula: Untrusted Re-Encryption Proxies
 
-```bash
-$ nucypher ursula run --dev --provider ipc:///tmp/geth.ipc --checksum-address <GETH_DEV_ADDRESS>
+When initializing an `Alice`, `Bob`, or `Ursula`, an initial "Stranger-`Ursula`" is needed to perform 
+the role of a `Teacher`, or "seednode":
+
+```python
+from nucypher.characters.lawful import Ursula, Alice
+
+ursula = Ursula.from_seed_and_stake_info(seed_uri=<URI>, federated_only=False)  # ie. https://0.0.0.0:9151
+another_ursula = Ursula.from_seed_and_stake_info(seed_uri=<URI>, federated_only=False)
+```
+
+Stranger `Ursula`s can be created by invoking the `from_seed_and_stake_info` method, then a `list` of `known_nodes`
+can be passed into any `Character`'s init. The `known_nodes` will inform your character of all of the nodes
+they know about network-wide, then kick-off the automated node-discovery loop:
+
+```python
+alice = Alice(known_nodes=[ursula, another_ursula], ...)
+```
+
+## Alice: Grant Access to a Secret
+
+```python
+from nucypher.characters.lawful import Alice, Bob, Ursula
+from nucypher.network.middleware import RestMiddleware
+from nucypher.blockchain.eth.registry import InMemoryContractRegistry
+
```

Comment:
Should we have a reactor thread?
```python
from twisted.internet import reactor
from threading import Thread
Thread(target=reactor.run, args=(False,)).start()
...
alice.start_learning_loop()
```
Wanna have Alice discovering Ursulas

Reply:
Still thinking about a graceful way to handle this part of the API.  I do not like the reactor thread on first sight.